src/foam/dao/MaterializedDAO.js

```diff
@@ -115,6 +125,25 @@ foam.CLASS({
         cmd.setIndex(new MaterializedDAOIndex(this));
 
         getSourceDAO().cmd(cmd);
+
+        String[] daoKeys = getObservedDAOs();
+        if (daoKeys.length != 0) {
+          for ( String daoKey : daoKeys ) {
+            DAO dao = (DAO) getX().get(daoKey);
+            var self = this;
+            if ( dao != null ) dao.listen(new Sink() {
+              public void put(Object obj, Detachable sub) {
+                getAdapter().onObservedDAOUpdate(self, obj);
```

Comment:
It might be useful to pass the 'daoKey as the second argument to onObservedDAOUpdate().